src/dfu/dfu_main.cpp

```diff
@@ -0,0 +1,1002 @@
+    YAML::Node config = YAML::LoadFile(yaml_file_full_path);
+    YAML::Node resource_ids = config;
+
+    for (const auto& node : config)
+    {
+        string resource_id_string = node.first.as<string>();
+        size_t startPos = resource_id_string.find('(');
+        size_t endPos = resource_id_string.find(')', startPos);
+        int resource_id = -1;
+        if (startPos != string::npos and endPos != string::npos)
+        {
+            resource_id = stoi(resource_id_string.substr(startPos + 1, endPos - startPos - 1), 0, 16);
+        }
+        if (resource_id_string.find("DFU_CONTROLLER_SERVICER_RESID") != string::npos)
+        {
+            if (verbose_mode) {
+                cout << "DFU_CONTROLLER_SERVICER_RESID is " << resource_id << endl;
+            }
+            dfu_controller_servicer_resid = resource_id;
+        }
+        YAML::Node dedicated_commands = node.second["dedicated_commands"];
+        if (dedicated_commands.IsSequence())
+        {
+            for (const auto& command_name : commandList)
+            {
+                add_command(dedicated_commands, command_name);
+            }
+        }
+    }
+    if (dfu_controller_servicer_resid == -1)
+    {
+        cerr << "Error: DFU_CONTROLLER_SERVICER_RESID not set. Check the YAML file: " << yaml_file_full_path << endl;
+        exit(HOST_APP_ERROR);
+    }
+}
+
+void check_verbose_mode(int * argc, char ** argv){
+    opt_t * use_opt = option_lookup("--verbose", options, num_options);
+    size_t index = argv_option_lookup(*argc, argv, use_opt);
+    if(index != 0)
+    {
+        cout << "Verbose mode enabled" << endl;
+        verbose_mode = 1;
+        remove_opt(argc, argv, index, 1);
+    }
+}
+
+uint16_t check_upload_start(int * argc, char ** argv){
+    opt_t * use_opt = option_lookup("--upload-start", options, num_options);
+    size_t index = argv_option_lookup(*argc, argv, use_opt);
+    // return -1 if the upload-start option is not found
+    if (index == 0) {
+        return INVALID_TRANSPORT_BLOCK_NUM;
+    }
+    if (index + 1 >= *argc)
+    {
+        cerr << "Error: missing block number" << endl;
+        exit(HOST_APP_ERROR);
+    }
+    string block_number_str = argv[index + 1];
+    if (!isdigit(block_number_str[0])) {
+        cerr << "Error: Value for transport block is not an integer. Given value: " << block_number_str << endl;
+        exit(HOST_APP_ERROR);
+    }
+    if (stoi(block_number_str) >= INVALID_TRANSPORT_BLOCK_NUM)
+    {
+        cerr << "Error: Max value for transport block is " << INVALID_TRANSPORT_BLOCK_NUM-1 << ". Given value: " << block_number_str << endl;
+        exit(HOST_APP_ERROR);
+    }
+    remove_opt(argc, argv, index, 2);
+
+    return stoi(block_number_str);
+}
+int main(int argc, char ** argv)
+{
+    if(argc == 1)
+    {
+        cout << "Use --help to get the list of options for this application." << endl
+        << "Or use --list-commands to print the list of commands and their info." << endl;
```

Comment:
As far as I see, you don't support --list-commands in this application

Reply:
Good spot. I removed that line.